include/dlaf/communication/kernels.h

```diff
@@ -68,16 +68,53 @@ matrix::Tile<const T, D> recvBcastAlloc(TileElementSize tile_size, comm::IndexT_
 template <class T, Device D, template <class> class Future>
 void scheduleSendBcast(const comm::Executor& ex, Future<matrix::Tile<const T, D>> tile,
                        hpx::future<common::PromiseGuard<comm::Communicator>> pcomm) {
-  hpx::dataflow(ex, hpx::util::unwrapping(sendBcast_o), internal::prepareSendTile(std::move(tile)),
+  hpx::dataflow(ex, matrix::unwrapExtendTiles(sendBcast_o), internal::prepareSendTile(std::move(tile)),
                 std::move(pcomm));
 }
 
+// This implements scheduleRecvBcast for when the device of the given tile and
+// the communication device are different. In that case we make use of
+// recvBcastAlloc, which will allocate a tile on the correct device for
```

Comment:
@msimberg @rasolca 

should we keep in mind that maybe this will in the next future accept a "workspace" tile from the user?

Reply:
From the user using dlaf algorithms or the user writing dlaf algorithms? This already takes a tile from the panel (which I assume you're referring to as a workspace). Are you thinking about something else?

Reply:
Sorry, it wasn't very clear what I meant. I should had pointed out these lines

https://github.com/eth-cscs/DLA-Future/blob/505e99981434189a06441f232ea3aa8d761920d6/include/dlaf/communication/kernels.h#L91-L93

where a temporary tile is allocated and used.

It is true that you are using it for the communication and then it is eventually copied back to the user passed one. I was just wondering if we can rely on the allocator (e.g. umpire) for these kind of temporary tiles or not.

Reply:
I see now, thanks for the clarification. Yes, that's exactly what's happening, and indeed, one of the main reasons to use umpire is that it has a pool of memory that does not need to synchronize the device for every allocation. It's not guaranteed to never synchronize, but the pool is initialized with a large block to make it less likely, the initial size is configurable, and if it has to grow the newly allocated memory will be reused and again make device synchronization less likely.

With CUDA 11.something we could use `cudaMallocAsync` to guarantee no device synchronization, but until then we need some pool, and for that I found umpire the best choice to avoid having to implement a memory pool in dlaf.